src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,193 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
+    public MyLinkedList() {
+    }
+
     @Override
     public void add(T value) {
+        Node<T> oldLast = last;
+        Node<T> newNode = new Node<>(oldLast, value, null);
+        last = newNode;
+        if (oldLast != null) {
+            oldLast.next = newNode;
+        } else {
+            first = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkPositionIndex(index);
+        if (index == size) {
+            add(value);
+        } else {
```

Comment:
```suggestion
        if (index == size) {
            add(value);
            return;
        }
```
prefer early return instead of indenting a lot of code, it's more readable